front/src/modules/pathfinding/hooks/usePathfinding.ts

```diff
@@ -228,193 +199,193 @@ export const usePathfinding = (
+    );
+
+    // We update existing pathsteps with coordinates, positionOnPath and kp corresponding to the new pathfinding result
+    const updatedPathSteps: (PathStep | null)[] = pathStepsInput.map((step, i) => {
+      if (!step) return step;
+      const correspondingOp = suggestedOperationalPoints.find((suggestedOp) =>
+        matchPathStepAndOp(step, suggestedOp)
       );
 
-      // We update existing pathsteps with coordinates, positionOnPath and kp corresponding to the new pathfinding result
-      const updatedPathSteps: (PathStep | null)[] = pathSteps.map((step, i) => {
-        if (!step) return step;
-        const correspondingOp = suggestedOperationalPoints.find((suggestedOp) =>
-          matchPathStepAndOp(step, suggestedOp)
-        );
+      const theoreticalMargin = i === 0 ? step.theoreticalMargin || '0%' : step.theoreticalMargin;
 
-        const theoreticalMargin = i === 0 ? step.theoreticalMargin || '0%' : step.theoreticalMargin;
+      const stopFor = i === pathStepsInput.length - 1 && !step.stopFor ? '0' : step.stopFor;
+      const stopType = i === pathStepsInput.length - 1 && !step.stopFor ? undefined : step.stopType;
 
-        const stopFor = i === pathSteps.length - 1 && !step.stopFor ? '0' : step.stopFor;
-        const stopType = i === pathSteps.length - 1 && !step.stopFor ? undefined : step.stopType;
+      return {
+        ...step,
+        positionOnPath: pathResult.path_item_positions[i],
+        stopFor,
+        stopType,
+        theoreticalMargin,
+        ...(correspondingOp && {
+          name: correspondingOp.name,
+          uic: correspondingOp.uic,
+          ch: correspondingOp.ch,
+          kp: correspondingOp.kp,
+          coordinates: correspondingOp.coordinates,
+        }),
+      };
+    });
+
+    if (!isEmptyArray(powerRestrictions)) {
+      dispatch(
+        setWarning({
+          title: t('warningMessages.pathfindingChange'),
+          text: t('warningMessages.powerRestrictionsReset'),
+        })
+      );
+    }
+    dispatch(updatePathSteps(updatedPathSteps));
+
+    const allWaypoints = upsertPathStepsInOPs(
+      suggestedOperationalPoints,
+      compact(updatedPathSteps)
+    );
+
+    setPathProperties({
+      electrifications,
+      geometry,
+      suggestedOperationalPoints,
+      allWaypoints,
+      length: pathResult.length,
+      trackSectionRanges: pathResult.track_section_ranges,
+      incompatibleConstraints,
+    });
+  };
 
-        return {
-          ...step,
-          positionOnPath: pathResult.path_item_positions[i],
-          stopFor,
-          stopType,
-          theoreticalMargin,
-          ...(correspondingOp && {
-            name: correspondingOp.name,
-            uic: correspondingOp.uic,
-            ch: correspondingOp.ch,
-            kp: correspondingOp.kp,
-            coordinates: correspondingOp.coordinates,
-          }),
-        };
-      });
+  const launchPathfinding = async (newPathSteps?: (PathStep | null)[]) => {
```

Comment:
Do we need `newPathSteps` to be optional here? Doesn't seem like any caller sets it to `undefined`.